flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java

```diff
@@ -1905,6 +2147,275 @@ public boolean filter(Event value) throws Exception {
 		));
 	}
 
+	@Test
+	public void testTimesRangeNonStrictOptional1() {
```

Comment:
The next two tests are slightly different from this one:
testTimesRangeNotStrictOption1: followedBy().times().optional().followedBy
testTimesRangeNotStrictOption2: followedByAny().times().allowCombinations().optional().followedBy()
testTimesRangeNotStrictOption3: followedByAny().times().optional().followedBy()